source/common/network/filter_state_dst_address.h

```diff
@@ -10,9 +10,10 @@ namespace Network {
 /**
  * Overrides the address selection for extensions, e.g. ORIGINAL_DST cluster.
  */
-class AddressObject : public StreamInfo::FilterState::Object, public Hashable {
+class AddressObject : public Address::InstanceAccessor, public Hashable {
 public:
-  AddressObject(Network::Address::InstanceConstSharedPtr address) : address_(address) {}
+  AddressObject(Network::Address::InstanceConstSharedPtr address)
+      : Address::InstanceAccessor(address), address_(address) {}
```

Comment:
It seems that the address is now kept in 2 places: the `address_` data member in `AddressObject` and the `ip_` data member in `InstanceAccessor`.
I suggest ensuring ODR by only keeping the data member in `InstanceAccessor`.